main/pipeline/pipeline.cc

```diff
@@ -226,6 +228,28 @@ unique_ptr<parser::Node> runParser(core::GlobalState &gs, core::FileRef file, co
     return nodes;
 }
 
+unique_ptr<parser::Node> runRBSRewrite(core::GlobalState &gs, core::FileRef file, unique_ptr<parser::Node> node,
+                                       const options::Printers &print) {
+    if (gs.cacheSensitiveOptions.rbsSignaturesEnabled || gs.cacheSensitiveOptions.rbsAssertionsEnabled) {
+        Timer timeit(gs.tracer(), "runRBSRewrite", {{"file", string(file.data(gs).path())}});
+        core::MutableContext ctx(gs, core::Symbols::root(), file);
+        core::UnfreezeNameTable nameTableAccess(gs);
+
+        if (gs.cacheSensitiveOptions.rbsSignaturesEnabled) {
+            auto rewriter = rbs::SigsRewriter(ctx);
+            node = rewriter.run(move(node));
+        }
+        if (gs.cacheSensitiveOptions.rbsAssertionsEnabled) {
+            auto rewriter = rbs::AssertionsRewriter(ctx);
+            node = rewriter.run(move(node));
+        }
+    }
+    if (print.RBSRewriteTree.enabled) {
+        print.ParseTree.fmt("{}\n", node->toStringWithTabs(gs, 0));
+    }
```

Comment:
Does it make sense to move this into the if block above? That way we could do some validation in options.cc to ensure that if we're asking to print the result of the rbs rewriter, we also have rbs enabled?

Reply:
Good idea, I also added a CLI test showing that it errors properly

Reply:
Yeah, good point. Let's keep it how it is 👍